python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -0,0 +1,217 @@
+
+        lrs = cyclic_learning_rate.ExponentialCyclicalLearningRate(
+            initial_learning_rate=0.1,
+            maximal_learning_rate=1.0,
+            step_size=5,
+            gamma=0.9,
+        )
+        scheduler = [lrs(i) for i in range(25)]
+        self.assertTrue(self.compare_lr_values(scheduler, gold))
+
+    def test_exp_lr(self):
+        """Test a exponential learning rate"""
+        gold = [0.10000000149011612,0.09486832469701767,0.08999999612569809,0.08538150042295456,0.08099999278783798,
+                0.07684334367513657,0.07289998978376389,0.06915900856256485,0.06560999155044556,0.062243103981018066,
+                0.05904899165034294,0.05601879581809044,0.05314409360289574,0.050416912883520126,0.04782968387007713,
+                0.045375220477581024,0.04304671287536621,0.04083769768476486,0.03874203935265541,0.036753926426172256,
+                0.03486783429980278,0.03307853266596794,0.03138105198740959,0.029770677909255028,0.028242945671081543]
+
+        lrs = ExponentialDecay(
+            initial_learning_rate=0.1,
+            decay_steps=2,
+            decay_rate=0.9,
+        )
+        scheduler = [lrs(i) for i in range(25)]
+        self.assertTrue(self.compare_lr_values(scheduler, gold))
+
+    def test_constant_lr(self):
+        """Test a constant learning rate"""
+        lrs = ExponentialDecay(
+            initial_learning_rate=0.1,
+            decay_steps=10000000,
+            decay_rate=1.0,
+        )
+        for i in range(25):
+            assert round(float(lrs(i).numpy()), 5) == 0.1
+
+    def test_cyclic_lr_in_training_pipeline(self):
+        """Test a cyclic learning rate in model training"""
+        Logger = logging_utils.setup_logging(
+            reset=True,
+            file_name=os.path.join(INPUT_DIR + 'ranklib_test_data', "output_log.csv"),
+            log_to_file=True,
+        )
+
+        io = LocalIO()
+        feature_config = self.parse_config(TFRecordTypeKey.SEQUENCE_EXAMPLE, self.feature_config_yaml_convert_to_clicks,
+                                           io)
+
+        dataset = RelevanceDataset(
+            data_dir=INPUT_DIR + '/ranklib_test_data',
+            data_format=DataFormatKey.RANKLIB,
+            feature_config=feature_config,
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            batch_size=2,
+            file_io=io,
+            preprocessing_keys_to_fns={},
+            logger=Logger,
+            keep_additional_info=KEEP_ADDITIONAL_INFO,
+            non_zero_features_only=NON_ZERO_FEATURES_ONLY,
+            max_sequence_size=319,
+        )
+
+        # Define interaction model
+        interaction_model: InteractionModel = UnivariateInteractionModel(
+            feature_config=feature_config,
+            feature_layer_keys_to_fns={},
+            tfrecord_type=TFRecordTypeKey.SEQUENCE_EXAMPLE,
+            max_sequence_size=319,
```

Comment:
@mohazahran Why is the `max_sequence_size` 319 here?

Reply:
Same question for above.

Reply:
Just use `ml4ir/applications/ranking/test/data/tfrecord` and set it to default.

Reply:
The sample dataset I use has a query with 319 docs

Reply:
Exactly. Change it to the one I pointed above.